src/subscription_manager/i18n.py

```diff
@@ -45,32 +45,40 @@ def configure_i18n():
     except locale.Error:
         # Following message could be little bit confusing. Why? When environment variable
         # LANG is set to e.g. es_ES.UTF-8, then we can show localized message for this language,
-        # but this language could not be fully supported by the system. It means that en_ES is not
+        # but this language could not be fully supported by the system. It means that es_ES is not
         # listed in the list of 'locale -a'
-        _locale = None
-        if "LANG" in os.environ:
-            _locale = os.environ["LANG"]
-        elif "LC_ALL" in os.environ:
-            _locale = os.environ["LC_ALL"]
+
+        # Determine the locale that setlocale attempted to set to use in the warning message.
+        # There is a cornercase if the env var that setlocale() raised an error for is neither
+        # LC_ALL nor LANG but we ignore that for now.
+        _locale = os.environ.get("LC_ALL", None)
+        if not _locale:
+            _locale = os.environ.get("LANG", None)
+
         if _locale is not None and Locale.is_locale_supported(_locale):
             print(
                 'You are attempting to use a locale: "%s" that is not fully supported by this system.'
                 % _locale
             )
+        ### Is this fine?  C.UTF-8 doesn't exist on RHEL7 so these can lead to exceptions.
```

Comment:
It is fine. We backport only security issues to RHEL-7 ATM.

Reply:
Excellent.  I will remove this comment then.